test/grouping.jl

```diff
@@ -3272,4 +3354,13 @@ end
     @test df == df2
 end
 
+@testset "invalid nthreads" begin
+    gdf = groupby(DataFrame(x=1:10, y=1:10), :y)
+    DataFrames.NTHREADS[] = 0
+    @test_throws ArgumentError select(gdf, :x => sum)
+    @test_throws ArgumentError transform(gdf, :x => sum)
+    @test_throws ArgumentError select!(gdf, :x => sum)
+    @test_throws ArgumentError transform!(gdf, :x => sum)
```

Comment:
make sure we do not leave an invalid value after the test
```suggestion
    @test_throws ArgumentError transform!(gdf, :x => sum)
    DataFrames.NTHREADS[] = 1
```